contracts/LinkedAccounts.cdc

```diff
@@ -0,0 +1,1082 @@
+
+/// This contract establishes a standard set of resources representing linked account associations, enabling
+/// querying of either end of account links as well as management of linked accounts, the Capabilities they're
+/// granted, as well as addition/removal of new/existing account delegations. By leveraging this contract, a new sort
+/// of custody is unlocked - Hybrid Custody - enabling the mainstream-friendly walletless onboarding UX we're so
+/// excited about on Flow.
+///
+/// By leveraging existing metadata standards, builders can easily query a Collection's linked accounts, their
+/// relevant metadata, etc. With implementation of the NFT standard, Collection owners can easily transfer delegation
+/// they wish in a mental model that's familiar and easy to understand.
+///
+/// The Collection allows a main account to add linked accounts, and An account is deemed a child of a
+/// parent if the parent maintains delegated access on the child account by way of AuthAccount
+/// Capability wrapped in an NFT and saved in a Collection. By
+/// the constructs defined in this contract, a linked account can be identified by a stored
+/// Handler.
+///
+/// While one generally would not want to share account access with other parties, this can be helpful in a low-stakes
+/// environment where the parent account's owner wants to delegate transaction signing to a secondary party. The idea 
+/// for this setup was born out of pursuit of a more seamless on-chain gameplay UX where a user could let a game client
+/// submit transactions on their behalf without signing over the whole of their primary account, and do so in a way
+/// that didn't require custom a Capability.
+///
+/// With that said, users should bear in mind that any assets in a linked account incur obvious custodial risk, and
+/// that it's generally an anti-pattern to pass around AuthAccounts. In this case, a user owns both accounts so they
+/// are technically passing an AuthAccount to themselves in calls to resources that reside in their own account, so 
+/// it was deemed a valid application of the pattern. That said, a user should be cognizant of the party with key
+/// access on the linked account as this pattern requires some degree of trust in the custodying party.
+///
+pub contract LinkedAccounts : NonFungibleToken, ViewResolver {
+
+    /// The number of NFTs in existence
+    pub var totalSupply: UInt64
+
+    // NFT conforming events
+    pub event ContractInitialized()
+    pub event Withdraw(id: UInt64, from: Address?)
+    pub event Deposit(id: UInt64, to: Address?)
+    
+    // LinkedAccounts Events
+    pub event MintedNFT(id: UInt64, parent: Address, child: Address)
+    pub event AddedLinkedAccount(parent: Address, child: Address, nftID: UInt64)
+    pub event UpdatedLinkedAccountParentAddress(previousParent: Address, newParent: Address, child: Address)
+    pub event UpdatedAuthAccountCapabilityForLinkedAccount(id: UInt64, parent: Address, child: Address)
```

Comment:
Not sure I follow - do you mean a more descriptive event name, param names, or data emitted in the event? As for the latter, there's not much other data to include since what's being updated for `UpdatedAuthAccountCapabilityForLinkedAccount` is the AuthAccount Capability for the specified child account in the NFT with given id and in parent's Collection. Is there something else you think would be helpful here?

Reply:
that makes sense. maybe the param name needs to be more descriptive, could `child` be changed to `newChildAddress` since the capability is changing?

Reply:
To clarify, the new AuthAccount Capability has to match the existing `linkedAccountAddress`, so the updated capability will be for the same child account Address. I thought to include the ability to update the wrapped AuthAccount Capability in the event the Capability needs to be retargeted for some reason.

On that note, maybe it would be better to rename the event since it appears to be misleading?